scripts/cleanup_additional_repos.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+#
+# cleanup additional YUM repository(typically for security updates)
+# The format of ADDITIONAL_YUM_REPOS is like "repo=patches-repo,name=Install patches,baseurl=http://amazonlinux.$awsregion.$awsdomain/xxxx,priority=10"
+# Multiple yum repo can be specified separated by ';'
+
+if [ -z "${ADDITIONAL_YUM_REPOS}" ]; then
+  echo "no additional yum repo, skipping"
+  exit 0
+fi
+
+
+AWK_CMD='
+BEGIN {RS=";";FS=","}
+{
+  delete vars; 
+  for(i = 1; i <= NF; ++i) { 
+    n = index($i, "="); 
+    if(n) { 
+      vars[substr($i, 1, n-1)] = substr($i, n + 1)
+    }
+  } 
+  Repo = "/etc/yum.repos.d/"vars["repo"]".repo"
+}
+{cmd="rm -f " Repo; system(cmd)}
+'
+echo "${ADDITIONAL_YUM_REPOS}" | awk "$AWK_CMD"
```

Comment:
You could use a here-string...
```sh
awk "$AWK_CMD" <<< "${ADDITIONAL_YUM_REPOS}"
```

Reply:
changed :D i don't even know it can be used like this 🐰